src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.cs

```diff
@@ -154,12 +154,6 @@ public static MemoryMappedFile CreateFromFile(string path, FileMode mode, string
                 throw new ArgumentException(SR.Argument_EmptyFile);
             }
 
-            if (access == MemoryMappedFileAccess.Read && capacity > fileStream.Length)
```

Comment:
If you are deferring this check, then you should missed to remove the condition in line 143, you also moved that one to `CreateCore`:
```cs
if (access == MemoryMappedFileAccess.Write)
{
    throw new ArgumentException(SR.Argument_NewMMFWriteAccessNotAllowed, nameof(access));
}
```